src/block-shapes.ts

```diff
@@ -0,0 +1,329 @@
+    MOTION_XSCROLL = 'motion_xscroll',
+    MOTION_YSCROLL = 'motion_yscroll',
+    OPERATOR_ADD = 'operator_add',
+    OPERATOR_SUBTRACT = 'operator_subtract',
+    OPERATOR_MULTIPLY = 'operator_multiply',
+    OPERATOR_DIVIDE = 'operator_divide',
+    OPERATOR_RANDOM = 'operator_random',
+    OPERATOR_LT = 'operator_lt',
+    OPERATOR_EQUALS = 'operator_equals',
+    OPERATOR_GT = 'operator_gt',
+    OPERATOR_AND = 'operator_and',
+    OPERATOR_OR = 'operator_or',
+    OPERATOR_NOT = 'operator_not',
+    OPERATOR_JOIN = 'operator_join',
+    OPERATOR_LETTER_OF = 'operator_letter_of',
+    OPERATOR_LENGTH = 'operator_length',
+    OPERATOR_CONTAINS = 'operator_contains',
+    OPERATOR_MOD = 'operator_mod',
+    OPERATOR_ROUND = 'operator_round',
+    OPERATOR_MATHOP = 'operator_mathop',
+    PROCEDURES_DEFINITION = 'procedures_definition',
+    PROCEDURES_CALL = 'procedures_call',
+    PROCEDURES_PROTOTYPE = 'procedures_prototype',
+    PROCEDURES_DECLARATION = 'procedures_declaration',
+    ARGUMENT_REPORTER_BOOLEAN = 'argument_reporter_boolean',
+    ARGUMENT_REPORTER_STRING_NUMBER = 'argument_reporter_string_number',
+    ARGUMENT_EDITOR_BOOLEAN = 'argument_editor_boolean',
+    ARGUMENT_EDITOR_STRING_NUMBER = 'argument_editor_string_number',
+    SENSING_TOUCHINGOBJECT = 'sensing_touchingobject',
+    SENSING_TOUCHINGOBJECTMENU = 'sensing_touchingobjectmenu',
+    SENSING_TOUCHINGCOLOR = 'sensing_touchingcolor',
+    SENSING_COLORISTOUCHINGCOLOR = 'sensing_coloristouchingcolor',
+    SENSING_DISTANCETO = 'sensing_distanceto',
+    SENSING_DISTANCETOMENU = 'sensing_distancetomenu',
+    SENSING_ASKANDWAIT = 'sensing_askandwait',
+    SENSING_ANSWER = 'sensing_answer',
+    SENSING_KEYPRESSED = 'sensing_keypressed',
+    SENSING_KEYOPTIONS = 'sensing_keyoptions',
+    SENSING_MOUSEDOWN = 'sensing_mousedown',
+    SENSING_MOUSEX = 'sensing_mousex',
+    SENSING_MOUSEY = 'sensing_mousey',
+    SENSING_SETDRAGMODE = 'sensing_setdragmode',
+    SENSING_LOUDNESS = 'sensing_loudness',
+    SENSING_LOUD = 'sensing_loud',
+    SENSING_TIMER = 'sensing_timer',
+    SENSING_RESETTIMER = 'sensing_resettimer',
+    SENSING_OF_OBJECT_MENU = 'sensing_of_object_menu',
+    SENSING_OF = 'sensing_of',
+    SENSING_CURRENT = 'sensing_current',
+    SENSING_DAYSSINCE2000 = 'sensing_dayssince2000',
+    SENSING_USERNAME = 'sensing_username',
+    SENSING_USERID = 'sensing_userid',
+    SOUND_SOUNDS_MENU = 'sounds_sounds_menu',
+    SOUND_PLAY = 'sound_play',
+    SOUND_PLAYUNTILDONE = 'sound_playuntildone',
+    SOUND_STOPALLSOUNDS = 'sound_stopallsounds',
+    SOUND_SETEFFECTTO = 'sound_seteffectto',
+    SOUND_CHANGEEFFECTBY = 'sound_changeeffectby',
+    SOUND_CLEAREFFECTS = 'sound_cleareffects',
+    SOUND_CHANGEVOLUMEBY = 'sound_changevolumeby',
+    SOUND_SETVOLUMETO = 'sound_setvolumeto',
+    SOUND_VOLUME = 'sound_volume',
+}
+
+export const BlockOpcodeToShape = {
+    [BlockOpcodes.CONTROL_FOREVER]: BlockShapes.CAP,
```

Comment:
Maybe not for this PR but we should turn these values into objects that might start like:

```js
{
  shape: BlockShapes.CAP
}
```

Reply:
shape will defenitely not be the only meta property for block opcodes - but we can refactor in next v